exercises/etl/canonical-data.json

```diff
@@ -0,0 +1,62 @@
+{
+  "transform": {
+    "description": "transforms the input map from a map of value: [ keys ] to a map of key: value for each of the keys",
```

Comment:
Despite having added the circular-buffer file, now I get to play devil's advocate: when the time comes to do #336, maybe someone will be unhappy that JSON files with only a single function-under-test have "cases" as a key under the top-level object, whereas those with multiple functions-under-test have "cases" in second-level objects.

But I have no particular reason to favour one way or the other until we come up with a proposed unified schema for all the JSON files.